### DISCUSSION TITLE
What will happen to secrets if I upgrade to v3?

### DISCUSSION THREAD

**Discussion by kotso** [netbox-community/netbox#7155] [Q&A] [answered]
Hello,

We are using net box for quite long and got couple hundreds of secrets stored in it.

What will happen after I migrate to version 3? 

I assume they will remain in database but will not be accessable from interface?

And is there any manual how to integrate Vault with Netbox? (maybe some options to export/migrate secrets there? as re-entering couple hundred of secrets will be one hell of the job)

**Comment by jeremystretch** [comment, marked as answer]
> I assume they will remain in database but will not be accessable from interface?

Correct. Barring any action beyond running the documented upgrade process, the records will remain the database. @DanSheps has been working on a [plugin](https://github.com/DanSheps/netbox-secretstore) to continue working with secrets on NetBox v3.0.

> And is there any manual how to integrate Vault with Netbox?

It's certainly possible but I'm not aware of any formal guides currently.

**Reply by DanSheps** [reply]
Yeah, unfortunately I need to do a few more tweaks as PIP doesn't install the templates.  Should hopefully have it fixed today.

**Reply by kotso** [reply]
@DanSheps if you can please write small guide for ppl who are running NetBox's under docker (docker-compose) from this repo (https://github.com/netbox-community/netbox-docker)
